src/server/game/Entities/Player/Player.cpp

```diff
@@ -10215,7 +10215,11 @@ void Player::SetRestBonus(float rest_bonus_new)
     if (rest_bonus_new < 0)
         rest_bonus_new = 0;
 
-    float rest_bonus_max = (float)GetUInt32Value(PLAYER_NEXT_LEVEL_XP) * 1.5f / 2;
+    // Fetch rest bonus multiplier from cached configuration
+    float rest_bonus_multiplier = sWorld->getRate(RATE_REST_MAX_MULTIPLIER);
```

Comment:
rename it to restBonusMultiplier (https://www.azerothcore.org/wiki/cpp-code-standards#naming-of-variables-and-functions)